### ISSUE TITLE
Should LocalDate format be 'date' instead of 'full-date'?

### ISSUE THREAD

**Issue by Tolyandre** [micro-elements/MicroElements.Swashbuckle.NodaTime#12]
Hi! I just looking for a way to setup Nodatime with Swashbuckle.AspNetCore 5.4.1 and found https://www.nuget.org/packages/MicroElements.Swashbuckle.NodaTime/3.0.0-rc.4 which worked for me.

I noticed that LocalDate produces format `full-date`. When I generate c# client back using nswag, it puts just string type for dates.

If I register a mapping directly with `date` format like
```c#
services.AddSwaggerGen(c =>
{
    c.SwaggerDoc("v1", new OpenApiInfo {Title = ApplicationName, Version = "v1"});
    c.MapType<LocalDate>(() => new OpenApiSchema { Type = "string", Format = "date" });               
    ...
});
```
then nswag will generate date type as expected.

As per the description, this library maps types to [ISO 8601](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14). Would it be better to use `date` for LocalDate format instead of `full-date` according to https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#data-types?

**Comment by jeremyhayes** [comment]
👍 on this.

The use of `full-date` as a format alias is not defined in the OpenAPI spec. In this case, it appears that the only supported aliases for date-related fields are
- `date-time`
- `date`

The OpenAPI spec references RFC3389 to describe the specific formatting indicated by these aliases, referring to differently-named aliases from the RFC. Presumably, these aliases have not been changed in the Swagger/OpenAPI spec for compatibility reasons.

In addition to the `full-date` alias discussed here, there are other `format` aliases being used for NodaTime types which should likely be reviewed at the same time:
- `full-date`
- `partial-time`
- `time-numoffset`

Rather than use unsupported `format` aliases, we might consider instead describing these values via `pattern`, using regexes to match the formats described in RFC3389. With `3.0.0` still in prerelease/candidate status, now may be a good time to consider these "breaking" changes. 🤔

References:
- OpenAPI 3.0.x spec, specifically for temporal types:
  - https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#data-types
- RFC3389, describing "Internet Date/Time Formats":
  - https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14
- Use of these undefined aliases, with reference to RFC3389:
  - https://github.com/micro-elements/MicroElements.Swashbuckle.NodaTime/blob/master/src/MicroElements.Swashbuckle.NodaTime/SchemasFactory.cs#L51

**Comment by petriashev** [comment]
closed by #13
